fern/assets/dashboard-referral.js

```diff
@@ -0,0 +1,84 @@
+/* eslint-disable no-shadow */
+!(function () {
+  "use strict";
+  const e = {
+    utm_source: "docs",
+    utm_medium: "referral",
+    utm_campaign: "docs_to_dashboard",
+  };
+  function t() {
+    const e = window.location.href,
+      t = window.location.pathname,
+      n = e.match(/\/docs\/(.+)$/);
+    return n
+      ? n[1].replace(/\//g, "_")
+      : t.replace(/^\//, "").replace(/\//g, "_") || "homepage";
+  }
+  function n(t, n) {
+    try {
+      const o = new URL(t);
+      return (
+        o.searchParams.set("utm_source", e.utm_source),
+        o.searchParams.set("utm_medium", e.utm_medium),
+        o.searchParams.set("utm_campaign", e.utm_campaign),
+        o.searchParams.set("utm_content", n),
+        o.toString()
+      );
+    } catch (e) {
+      return console.warn("Error adding UTM parameters to URL:", t, e), t;
+    }
+  }
+  function o() {
+    const e = t();
+    let o = 0;
+    document
+      .querySelectorAll('a[href*="dashboard.alchemy.com"]')
+      .forEach((t) => {
+        const r = t.getAttribute("href");
+        if (r && r.match(/^https?:\/\/dashboard\.alchemy\.com/)) {
+          const s = n(r, e);
+          t.setAttribute("href", s),
+            o++,
+            t.setAttribute("data-utm-tracked", "true");
+        }
+      }),
+      o > 0;
+  }
+  function r() {
+    o();
+    const e = new MutationObserver(function (e) {
```

Comment:
Out of curiosity, what is this needed for? Does the script not work on page changes without it?